src/Concerns/HasRevisions.php

```diff
@@ -229,7 +229,7 @@ public function getNewestIdAttribute($value)
             return $value;
         }
         // dependency on latest boolean column, alternative to using max id
-        return $this->revisions()->where('latest', true)->first()->revisionable_id;
+        return $this->revisions()->where('latest', true)->first()->revisionable_id ?? null;
```

Comment:
If you are going through the trouble of checking if it is set as an attribute before computation, should you not set the attribute with the result before returning. I don't believe access mutators persist the attribute?

The other concern I would have about the attribute existing on the model, is that if you went to save a model with this attribute set, would it not trigger an error since it is not in the database? Or is this being explicitly exluded somewhere else?

Reply:
To exemplify:

In getNewestIdAttribute:
```php
    /**
     * Get the newest revision in the lineage
     *
     * @return static
     */
    public function getNewestIdAttribute($value)
    {
        if ($value !== null || array_key_exists('newest_id', $this->attributes)) {
            return $value;
        }
        dump('computing...');
        // dependency on latest boolean column, alternative to using max id
        return $this->revisions()->where('latest', true)->first()->revisionable_id ?? null;
    }
```

```php
        $post = factory(Post::class)->create();

        $post->newest_id;
        $post->newest_id;
        $post->newest_id;
        $post->newest_id;

        dd('Stop!');

        Output:
        ^ "computing..."
        ^ "computing..."
        ^ "computing..."
        ^ "computing..."
        ^ "Stop!"
```

```php
        $post = factory(Post::class)->create();

        $post = Post::whereKey($post->id)->withNewest()->first();

        $post->newest_id;
        $post->newest_id;
        $post->newest_id;
        $post->newest_id;

        dd('Stop!');
        
        Output:
        ^ "Stop!"
```

Reply:
Also, re the second point.

The following:
```php
        $post = factory(Post::class)->create();

        $post = Post::whereKey($post->id)->withNewest()->first();

        $post->title = 'Changed';
        $post->save();
```

Throws:
```
Illuminate\Database\QueryException: SQLSTATE[HY000]: General error: 1 table posts has no column named initial_id
```

Reply:
Looks like I have to rethink the whole dynamic relationship ordeal. I generally use it for pulling data and not saving. I support we'd have to explicitly unset it somewhere in the save process. going to recheck that eloquent course I had for some inspiration. 

Re: the repeating computation, if we can figure you the 2nd part of saving then setting it directly would be a feasible solution.

Reply:
Yes. I feel like there must be (or should be) a way to specify attributes you do not want saved on the model.

Reply:
Short of modifying `Model::getAttributes()` to explicitly ignore these columns I can't see a proper way of handling it.

re: the mutator not caching the query, I've opted to use a relation from revision, this will at least limit us to 2 queries as the relations will be cached